test/cri-containerd/container_update_test.go

```diff
@@ -0,0 +1,379 @@
+// +build functional
+
+package cri_containerd
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/Microsoft/hcsshim/osversion"
+	testutilities "github.com/Microsoft/hcsshim/test/functional/utilities"
+	runtime "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+)
+
+const processorWeightMax = 10000
+
+func calculateJobCPUWeight(processorWeight uint32) uint32 {
+	if processorWeight == 0 {
+		return 0
+	}
+	return 1 + uint32((8*processorWeight)/processorWeightMax)
+}
+
+func calculateJobCPURate(hostProcs uint32, processorCount uint32) uint32 {
+	rate := (processorCount * 10000) / hostProcs
+	if rate == 0 {
+		return 1
+	}
+	return rate
+}
+
+func Test_Container_UpdateResources_CPUShare(t *testing.T) {
+	testutilities.RequiresBuild(t, osversion.V20H2)
+	type config struct {
+		name             string
+		requiredFeatures []string
+		runtimeHandler   string
+		sandboxImage     string
+		containerImage   string
+		cmd              []string
+	}
+	tests := []config{
+		{
+			name:             "WCOW_Process",
+			requiredFeatures: []string{featureWCOWProcess},
+			runtimeHandler:   wcowProcessRuntimeHandler,
+			sandboxImage:     imageWindowsNanoserver,
+			containerImage:   imageWindowsNanoserver,
+			cmd:              []string{"cmd", "/c", "ping", "-t", "127.0.0.1"},
+		},
+		{
+			name:             "WCOW_Hypervisor",
+			requiredFeatures: []string{featureWCOWHypervisor},
+			runtimeHandler:   wcowHypervisorRuntimeHandler,
+			sandboxImage:     imageWindowsNanoserver,
+			containerImage:   imageWindowsNanoserver,
+			cmd:              []string{"cmd", "/c", "ping", "-t", "127.0.0.1"},
+		},
+		{
+			name:             "LCOW",
+			requiredFeatures: []string{featureLCOW},
+			runtimeHandler:   lcowRuntimeHandler,
+			sandboxImage:     imageLcowK8sPause,
+			containerImage:   imageLcowAlpine,
+			cmd:              []string{"top"},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			requireFeatures(t, test.requiredFeatures...)
+
+			if test.runtimeHandler == lcowRuntimeHandler {
+				pullRequiredLcowImages(t, []string{test.sandboxImage})
+			} else if test.runtimeHandler == wcowHypervisorRuntimeHandler {
```

Comment:
I had the same comment in another test somewhere but we're only pulling the windows image if the hypervisor runtime is specified and not for process with this